components/soap/src/main/java/org/switchyard/component/soap/composer/SOAPContextMapper.java

```diff
@@ -90,11 +102,48 @@ public void mapFrom(SOAPBindingData source, Context context) throws Exception {
         if (source.getStatus() != null) {
             context.setProperty(HTTP_RESPONSE_STATUS, source.getStatus()).addLabels(SOAP_MIME_LABELS);
         }
+        
+        //Gets the propagation label from the service domain
+        String propagationLabel=null;
+        if(context.getProperty(SERVICE_REFERENCE_PROPERTY)!=null){
+            ServiceDomain domain= ((ServiceReference)context.getProperty(SERVICE_REFERENCE_PROPERTY).getValue()).getDomain();
+            if(domain.getProperty(PropertyConstants.DOMAIN_PROPERTY_PROPAGATE_LABEL)!=null){
+            	propagationLabel=(String)domain.getProperty(PropertyConstants.DOMAIN_PROPERTY_PROPAGATE_LABEL);
```

Comment:
This was talk with Tom. The labels that are marked to be propagated are defined by a switchyard property. You can see all the example I created.